build.gradle

```diff
@@ -142,19 +143,54 @@ tasks.withType(JavaCompile).configureEach {
 	options.compilerArgs.add '-XDstringConcat=inline'
 }
 
-tasks.named("build").configure { dependsOn('installLocalGitHookPostCheckout') }
-tasks.register('installLocalGitHookPostCheckout', Copy) {
-	group = 'git'
-	from new File(rootProject.rootDir, '.scripts/post-checkout')
-	into { new File(rootProject.rootDir, '.git/hooks') }
-	filePermissions {
-		unix(0775)
+def getRootRelativePath(path) {
+	rootDir.toPath().relativize(path)
+}
+
+def ensureSymlink(link, target) {
+	if (!Files.isSymbolicLink(link)) {
+		Files.deleteIfExists(link)
+
+		Files.createSymbolicLink(link, target)
+
+		println("--- Installed symbolic link ${getRootRelativePath(link)} -> ${getRootRelativePath(target)} ---")
 	}
+}
+
+def gitConfig = 'Git Configuration'
+tasks.register('installLocalGitHookSymlinks') {
+	group = gitConfig
+
+	inputs.dir "$rootDir/.scripts/git-hooks/"
+	inputs.dir "$rootDir/.git/hooks/"
+	outputs.dir "$rootDir/.git/hooks/"
```

Comment:
consider to save the two paths as variables instead of duplicating it 5 times (another 2 times below)